src/EventLoop.cpp

```diff
@@ -282,12 +281,14 @@ auto EventLoop::checkActions(
             CURLcode easy_result = message->data.result;
 
             // So the reacquired shared_ptr destructs itself correctly, let's get it into a unique pointer
-            std::unique_ptr<std::shared_ptr<SharedRequest>> shared_request_ptr_pointer;
-            curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, &shared_request_ptr_pointer);
+            std::shared_ptr<SharedRequest>* curl_info_private_pointer{nullptr};
+            curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, &curl_info_private_pointer);
```

Comment:
What if this fails?

Reply:
Good point! No idea why we weren't checking the return code before...

Also added a check that curl_info_private_pointer is not a nullptr after the call to curl_easy_getinfo, since that would also be bad.